ehcache/src/main/java/com/c8y/ms/templates/ehcache/service/InventoryApiCachedService.java

```diff
@@ -0,0 +1,46 @@
+package com.c8y.ms.templates.ehcache.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import com.cumulocity.model.idtype.GId;
+import com.cumulocity.rest.representation.inventory.ManagedObjectRepresentation;
+import com.cumulocity.sdk.client.inventory.InventoryApi;
+
+@Service
+public class InventoryApiCachedService {
+	private static final Logger LOG = LoggerFactory.getLogger(InventoryApiCachedService.class);
+	
+	private InventoryApi inventoryApi;
+	
+	public InventoryApiCachedService(InventoryApi inventoryApi) {
+		super();
+		this.inventoryApi = inventoryApi;
+	}
+	
+	@Cacheable("managedObjects")
+	public ManagedObjectRepresentation get(String id) {
+		LOG.info("load ManagedObject of: {}", id); 
+		try {
+			ManagedObjectRepresentation managedObjectRepresentation = inventoryApi.get(GId.asGId(id));
+			if(managedObjectRepresentation == null) {
```

Comment:
I think this check might unnecessary as ManagedObjectRepresentation can't be null at this point as far as I know. If a managedObject couldn't be found an Exception will be returned.

Reply:
yes this is correct, i remove this null check